plugins/vpc-bridge/config/k8s_connector.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+//go:build enablek8sconnector && windows
+
+package config
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/amazon-vpc-cni-plugins/network/vpc"
+	"github.com/aws/amazon-vpc-cni-plugins/plugins/vpc-bridge/config/k8s"
+)
+
+var (
+	retrievePodConfigHandler = retrievePodConfig
+)
+
+// retrievePodConfig retrieves a pod's configuration from an external source.
+func retrievePodConfig(netConfig *NetConfig) error {
+
+	kc := netConfig.Kubernetes
+
+	// create context with timeout.
+	ctx := context.Background()
+	ctx, _ = context.WithTimeout(ctx, time.Minute)
+
+	// Get new k8s connector object to get pod IP.
+	podRetriever, err := k8s.NewK8sConnector()
+	if err != nil {
+		return fmt.Errorf("error creating k8s connector: %w", err)
+	}
+
+	// create pod object with pod namespace and pod name.
+	pod := k8s.Pod{
+		kc.Namespace,
+		kc.PodName,
+	}
+	// retrieve Pod IP Address CIDR string using k8s connector.
+	ipAddress, err := podRetriever.GetPodIP(ctx, pod)
```

Comment:
You dont need the pod struct. Just pass namespace, podName to getPodIP

Reply:
removed `pod` struct and started using `podName` and `podNamespace` directly.